app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserMenuSignIn.kt

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.item.BrowserMenuImageText
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+
+/**
+ * A menu item for displaying account information. The item calculates the label on every bind call,
+ * to provide each menu with the latest account manager state.
+ *
+ * @param label The function that provides the visible label of this menu item.
+ * @param imageResource ID of a drawable resource to be shown as icon.
+ * @param textColorResource Optional ID of color resource to tint the text.
+ * @param listener Callback to be invoked when this menu item is clicked.
+ */
+
+class BrowserMenuSignIn(
```

Comment:
I am curious about the rationale about this class since it doesn't look like it is doing anything extra that `BrowserMenuImageText` couldn't. What's strange to me was that we are passing a `label` param which will effectively not be utilized when we also just bind by the text view with `getLabel` anyways. So, I wouldn't expose `label` either as a param and just add the default `String()` value in line 35 if this class was needed.

It looks possible to me that we could just have the `getLabel` (maybe rename to `getSignInLabel`) function in `DefaultToolbarMenu.kt` and call it via

```
BrowserMenuImageText(
                    getSignInLabel(),
                    R.drawable.ic_signed_out,
                    primaryTextColor(),
                ) {
                    onItemTapped.invoke(
                        ToolbarMenu.Item.SyncAccount(accountManager.accountState),
                    )
                }
```

Reply:
My though process is described in the body of the PR. Also, it looks like Petru asked the same [question](https://github.com/mozilla-mobile/fenix/pull/28409#discussion_r1064500811).

The version that you suggested will calculate the `label` once in the constructor, and apply the text every `bind` call. In contrast, `BrowserMenuSignIn` will calculate it on every bind call (to support slow internet or opening the menu while `accountManager` is still busy).

You are right about passing the label parameter, it's a bit I forgot to remove. Was trying to write a test that would make sure that the value from constructor gets `overriden`, but got stuck trying to mock `onBind()` method.